lib/mhv/client.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+require 'faraday'
+require 'multi_json'
+require 'common/client/errors'
+require 'mhv/configuration'
+require 'mhv/api/new_users'
+
+module MHV
+  # Core class responsible for api interface operations
+  class Client
+    HOST = 'https://www.myhealth.va.gov'
+    REQUEST_TYPES = %i(post).freeze
+    USER_AGENT = 'Vets.gov Agent'
+    BASE_REQUEST_HEADERS = {
+      'Accept' => 'text/html',
+      'Content-Type' => 'application/x-www-form-urlencoded',
+      'User-Agent' => USER_AGENT
+    }.freeze
+
+    attr_reader :config
+
+    def initialize
+      @config = MHV::Configuration.new(host: HOST)
+      raise ArgumentError, 'config is invalid' unless @config.is_a?(Configuration)
+    end
+
+    private
+
+    def perform(method, path, params = nil, headers = nil)
+      raise NoMethodError, "#{method} not implemented" unless REQUEST_TYPES.include?(method)
```

Comment:
Is there a benefit to this check and raising this error here rather than letting Ruby raise it on its own on the next line? It doesn't look like `method` is user input so there shouldn't be any worry about calling a private or unexpected method from `send`